[2025-09-15T10:19:26.472+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T09:47:50.933492+00:00 [queued]>
[2025-09-15T10:19:26.479+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T09:47:50.933492+00:00 [queued]>
[2025-09-15T10:19:26.479+0000] {taskinstance.py:1361} INFO - Starting attempt 6 of 6
[2025-09-15T10:19:26.642+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): gold_saved_transform> on 2025-09-15 09:47:50.933492+00:00
[2025-09-15T10:19:26.648+0000] {standard_task_runner.py:57} INFO - Started process 912 to run task
[2025-09-15T10:19:26.651+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spotify_saved_tracks_etl_pipeline', 'gold_saved_transform', 'manual__2025-09-15T09:47:50.933492+00:00', '--job-id', '875', '--raw', '--subdir', 'DAGS_FOLDER/spotify_saved_dag.py', '--cfg-path', '/tmp/tmpgbs0ep_f']
[2025-09-15T10:19:26.653+0000] {standard_task_runner.py:85} INFO - Job 875: Subtask gold_saved_transform
[2025-09-15T10:19:26.685+0000] {task_command.py:415} INFO - Running <TaskInstance: spotify_saved_tracks_etl_pipeline.gold_saved_transform manual__2025-09-15T09:47:50.933492+00:00 [running]> on host 537418afed42
[2025-09-15T10:19:26.749+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_saved_tracks_etl_pipeline' AIRFLOW_CTX_TASK_ID='gold_saved_transform' AIRFLOW_CTX_EXECUTION_DATE='2025-09-15T09:47:50.933492+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-15T09:47:50.933492+00:00'
[2025-09-15T10:19:26.760+0000] {logging_mixin.py:151} INFO - Executed sql/gold/saved_tracks/fact_saved_songs.sql
[2025-09-15T10:19:26.765+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/etl/db_utils.py", line 11, in run_sql_scripts
    cur.execute(sql)
psycopg2.errors.DatatypeMismatch: column "song_duration_sec" is of type numeric but expression is of type text
LINE 9:     artist_id,
            ^
HINT:  You will need to rewrite or cast the expression.

[2025-09-15T10:19:26.772+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spotify_saved_tracks_etl_pipeline, task_id=gold_saved_transform, execution_date=20250915T094750, start_date=20250915T101926, end_date=20250915T101926
[2025-09-15T10:19:26.779+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 875 for task gold_saved_transform (column "song_duration_sec" is of type numeric but expression is of type text
LINE 9:     artist_id,
            ^
HINT:  You will need to rewrite or cast the expression.
; 912)
[2025-09-15T10:19:26.792+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-09-15T10:19:26.804+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
